File: backend/src/main/java/com/interview/platform/mod05_experience_level_management/repository/ExperienceLevelRepository.java
package com.interview.platform.mod05_experience_level_management.repository;

import com.interview.platform.mod05_experience_level_management.model.ExperienceLevelEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.UUID;

import java.util.Optional;

public interface ExperienceLevelRepository extends JpaRepository<ExperienceLevelEntity, UUID> {
    Optional<ExperienceLevelEntity> findByLabel(String label);
}

File: android-app/app/src/main/java/com/interview/platform/domain/repository/ExperienceLevelRepository.kt
package com.interview.platform.domain.repository

import com.interview.platform.domain.model.mod05_experience_level_management.ExperienceLevel
import com.interview.platform.data.model.Result
import kotlinx.coroutines.flow.Flow

interface ExperienceLevelRepository {
    fun fetchAll(): Flow<Result<List<ExperienceLevel>>>
    suspend fun syncData(): Result<Unit>
}
